lib/autoproj/daemon/git_api/pull_request.rb

```diff
@@ -85,6 +96,23 @@ def draft?
                 def mergeable?
                     @model["mergeable"]
                 end
+
+                # Resolve the list of recursive dependencies for this pull request
+                #
+                # @return [Set<GitAPI::PullRequest>]
+                def recursive_dependencies
+                    result = []
+                    queue = dependencies.dup
+                    until queue.empty?
+                        pr = queue.shift
+                        next if result.include?(pr) || pr == self
+
+                        result << pr
+                        queue.concat(pr.dependencies)
```

Comment:
Actually, shouldn't it be 

```ruby
                        queue.concat(pr.recursive_dependencies)
```

?

Reply:
Ah, doesn't matter, you add them to the queue. Nevermind.

Reply:
That would be doing actual programming recursion, and we would have to pass the "seen" set to the method to avoid infinite loops. In these cases (graphs with loops), I usually prefer doing a breadth-first search (which I'm doing here, that's what the `queue` is here for). I find that clearer personally.

Reply:
Yep, agreed.